### ISSUE TITLE
Add ffmpeg bundle

### ISSUE THREAD

**Issue by fabiohl** [clearlinux/clr-bundles#62]
Clear is great choice nor only for server or desktops, but also for automated build machines. For example, a spool of encoding videos, compiling executables, and others. Its focus on performance would a natural choice for encoding multimedia.
So I strongly suggest a bundle for ffmpeg and all codecs.

**Comment by OpenSourceAnarchist** [comment]
## :+1:

**Comment by ahkok** [comment]
Blocked.

**Comment by ahkok** [comment]
An alternative is to use flatpak and install `io.gnome.GnomeMPV`.

**Comment by OpenSourceAnarchist** [comment]
@ahkok That is great for playing videos/audio, but what about browser functionality? Last I checked on Clear Linux, h.264 codec isn't supported in Firefox. I'd be fine with running Chrome or Vivaldi or Opera, but I am not sure what the best way to install Chrome is and ffmpeg codecs are required for the others.

Really love Clear Linux but I use it only as a desktop. Functionality isn't there yet. Still can't decide between Arch, Void, Solus, etc.

**Comment by fabiohl** [comment]
Besides that, It would be enjoyable a native binary made with the standards of optimization we love in clear.

**Comment by miguelinux** [comment]
A guy from the team is working on a solution, but that task is low
priority and don't have ETA for it.

**Comment by OpenSourceAnarchist** [comment]
Could there at least be a guide on how to compile ffmpeg locally with proprietary codecs for Clear? That way users can still benefit but the distro doesn't have to deal with licenses, redistribution, etc.

**Comment by ahkok** [comment]
There is nothing special about compiling ffmpeg on clearlinux. The last time I did it I just downloaded it and did `configure` and `make`.

**Comment by OpenSourceAnarchist** [comment]
That makes sense. I figured Clear's unique packaging solution would make things more complicated, but if I can compile it locally without any problems, then that should resolve my usability problems with Clear. I still think some documentation about this should be easily accessible, even if it is just "we don't have the license to redistribute ffmpeg with proprietary codecs, so users can download x.tar.gz from (insert proper url) and compile locally by performing a standard configure, make, and make install. This will resolve usability issues on Firefox and playing some videos and music on Clear."

**Comment by rbharathkumar** [comment]
Thank you for your response ahkok. Appreciate your work.
As @OpenSourceAnarchist has mentioned, would love to have some kind of documentation available atleast the appropriate URL if available. Browser functionality is afterall a basic requirement.
Thanks again.

**Comment by gabrielgiordano** [comment]
I'm trying to compile ffmpeg to install shared libraries needed by firefox using the following command:

```
./configure --enabled-shared
make
```

But I'm getting this error:
```
LD      libavcodec/libavcodec.so.58
/usr/bin/ld: libavcodec/mqc.o: relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: libavcodec/mqcdec.o: relocation R_X86_64_32S against undefined symbol `ff_mqc_qe' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: libavcodec/svq13.o: relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: nonrepresentable section on output
collect2: error: ld returned 1 exit status
make: *** [ffbuild/library.mak:103: libavcodec/libavcodec.so.58] Error 1
```

Searching the internet for solutions I found some stackoverflow answers saying that this error probably means that some library that ffmpeg depends upon was not compiled with --enabled-shared. But I'm not sure how to find which library I need to compile here.

I got the source from here: https://github.com/FFmpeg/FFmpeg

Did any of you manage to compile ffmpeg with shared libraries successfully and got Firefox working with it as well?

**Comment by rbharathkumar** [comment]
I have not been able to get FFmpeg to work.
So after I download, configure, make and make install, videos play on firefox, but then again after the next update it breaks. So went ahead and installed Google Chrome via rpm. Everything works as expected. Hope it helps.

**Comment by gabrielgiordano** [comment]
Thank you for your answer @rbharathkumar . I've installed chrome following this script here: https://github.com/clearlinux/clr-bundles/issues/83#issuecomment-440205522 and now I can switch to chrome whenever firefox can't handle something.

**Comment by eero-t** [comment]
Isn't this duplicate of #43 ?

**Comment by rbharathkumar** [comment]
> Isn't this duplicate of #43 ?

Yes and I think this issue is also linked somewhere in that Issue itself.
